homeassistant/components/frontend/__init__.py

```diff
@@ -44,11 +51,28 @@
 
 DATA_PANELS = 'frontend_panels'
 DATA_INDEX_VIEW = 'frontend_index_view'
+DATA_THEMES = 'frontend_themes'
+DATA_CURRENT_THEME = 'frontend_current_theme'
+DEFAULT_THEME = 'default'
 
 # To keep track we don't register a component twice (gives a warning)
 _REGISTERED_COMPONENTS = set()
 _LOGGER = logging.getLogger(__name__)
 
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(ATTR_THEMES): vol.Schema({
+            cv.string: OrderedDict
```

Comment:
Do we want to whitelist the variables that a user can set? `primary-color`, `accent-color`. For values, we should also validate it's a string.

```python
          cv:string: {vol.In(THEME_KEYS): cv.string}
```